tests/integration/buildcmd/test_build_cmd.py

```diff
@@ -492,389 +263,36 @@ def _validate_skipped_built_function(
-    ("template",),
-    [
-        ("esbuild_templates/template_with_metadata_node_options.yaml",),
-        ("esbuild_templates/template_with_metadata_global_node_options.yaml",),
-    ],
-)
-class TestBuildCommand_EsbuildFunctionProperties(BuildIntegEsbuildBase):
-    @parameterized.expand(
-        [
-            ("nodejs16.x", "../Esbuild/TypeScript", "app.lambdaHandler", "x86_64"),
-            ("nodejs18.x", "../Esbuild/TypeScript", "app.lambdaHandler", "x86_64"),
-            ("nodejs20.x", "../Esbuild/TypeScript", "app.lambdaHandler", "x86_64"),
-            ("nodejs16.x", "../Esbuild/TypeScript", "nested/function/app.lambdaHandler", "x86_64"),
-            ("nodejs18.x", "../Esbuild/TypeScript", "nested/function/app.lambdaHandler", "x86_64"),
-            ("nodejs20.x", "../Esbuild/TypeScript", "nested/function/app.lambdaHandler", "x86_64"),
-        ]
-    )
-    def test_environment_generates_sourcemap(self, runtime, code_uri, handler, architecture):
-        overrides = {
-            "runtime": runtime,
-            "code_uri": code_uri,
-            "handler": handler,
-            "architecture": architecture,
-        }
-        self._test_with_various_properties(overrides, runtime)
-
-
-class TestBuildCommand_NodeFunctions_With_Specified_Architecture(BuildIntegNodeBase):
-    template = "template_with_architecture.yaml"
-
-    @parameterized.expand(
-        [
-            ("nodejs16.x", False, "x86_64"),
-            ("nodejs18.x", False, "x86_64"),
-            ("nodejs20.x", False, "x86_64"),
-            ("nodejs16.x", "use_container", "x86_64"),
-            ("nodejs18.x", "use_container", "x86_64"),
-        ]
-    )
-    def test_building_default_package_json(self, runtime, use_container, architecture):
-        if use_container and not runtime_supported_by_docker(runtime):
-            self.skipTest(RUNTIME_NOT_SUPPORTED_BY_DOCKER_MSG)
-        self._test_with_default_package_json(runtime, use_container, self.test_data_path, architecture)
+@pytest.mark.ruby
+class TestBuildCommand_RubyFunctions(BuildIntegRubyBase):
+    @parameterized.expand([(False,), ("use_container",)])
+    def test_building_ruby_3_2(self, use_container):
+        if use_container and SKIP_DOCKER_TESTS or SKIP_DOCKER_BUILD:
+            self.skipTest(SKIP_DOCKER_MESSAGE)
 
+        self._test_with_default_gemfile("ruby3.2", use_container, "Ruby", self.test_data_path)
 
-class TestBuildCommand_RubyFunctions(BuildIntegRubyBase):
-    @parameterized.expand(["ruby3.2", "ruby3.3"])
+    @parameterized.expand([("ruby3.3",)])
     @skipIf(SKIP_DOCKER_TESTS or SKIP_DOCKER_BUILD, SKIP_DOCKER_MESSAGE)
-    def test_building_ruby_in_container(self, runtime):
-        if IS_WINDOWS and not runtime_supported_by_docker(runtime):
-            self.skipTest(RUNTIME_NOT_SUPPORTED_BY_DOCKER_MSG)
+    @pytest.mark.al2023
+    def test_building_ruby_al2023(self, runtime):
         self._test_with_default_gemfile(runtime, "use_container", "Ruby", self.test_data_path)
```

Comment:
Aren't we removing `ruby3.2` from this tests?

Reply:
No, we since we only have 1 runtime in the old non-al2023 tests, the test_building_ruby_3_2 just directly calls ruby3.2 in the runtime so it is still the same test, just not parametrized